### ISSUE TITLE
Create Account Number endpoint 

### ISSUE THREAD

**Issue by JanellC** [US-EPA-CAMD/PI-1#1348]
### Definition of Ready
- [x] Does this meet the INVEST criteria?
- [x] Does this have a clearly defined user?
- [x] Does this have an estimate?
- [x] Does this have acceptance criteria?
- [x] Does this have sufficient context (wireframes, business rules, etc.)?
- [x] Is this traceable to a feature, epic, or PI?

## User Story
As a CAMPD API user I want to see a list of account numbers so that I can include the correct data in my query.

## Acceptance Criteria

Given that a user has made a GET request to the account numbers endpoint in the Account API
When the request is executed
Then the system displays all account numbers in JSON format and for each record displays:

- Account number

- Account name 

## Context
(User journeys, design mockups, architectural diagrams, business rules, data element definitions)
Use account fact 

ACCOUNT_FACT is the best place to get a complete list of accounts.  

ACCOUNT_FACT contains one record for each unique account+program combination.  

On CAMDPUB, it is loaded using the combined data from ACCOUNT_SS (CAMDBSP_CAMDAMS.ACCOUNT) and NTACCOUNT_SS (CAMDBSP_CAMDNATS.TACCOUNT).

Mark

From: Lageman, Travis <Lageman.Travis@epa.gov> 
Sent: Wednesday, May 12, 2021 4:49 PM
To: Ann Albrecht <Ann.Albrecht@erg.com>; Mark Hayward <Mark.Hayward@erg.com>
Cc: Stetsyuk, Karina <Stetsyuk.Karina@epa.gov>; Gill, Sandeep <Gill.Sandeep@epa.gov>; Desantis, Laurel <Desantis.Laurel@epa.gov>
Subject: Question about Account tables

CAUTION: Don't open links or attachments unless you recognize the sender and know they are safe.
Hi Mark & Ann,

Apologies for the double email, but thought it was easier to keep these two questions separate. Another question that arose with CVP today was which table is most appropriate to pull all account numbers/names? I am most familiar with ACCOUNT_FACT in CAMDPUB, which has accounts and their programs. 

I am assuming that CAMDBSP_CAMDAMS.ACCOUNT is for all programs except NBP and OTC (which I think is similar to a transaction related question I asked weeks ago), and that CAMDBSP_CAMDNATS.TACCOUNT is only for NBP and OTC programs.

Therefore, is ACCOUNT_FACT the best place to ensure that all accounts/numbers are collected?

Thanks again!

Travis Lageman
Environmental Policy Analyst
Clean Air Markets Division
U.S. Environmental Protection Agency
P: 202-564-0520

## Testing
Test Case: [TC168](https://cvpcorp.testquality.com/project/11428/plan/20060/test/280935)
Tester: @JanellC 

### Definition of Done
- [x] Have all Acceptance Criteria been agreed to and validated by the scrum team?
- [x] Has the code been unit tested and peer reviewed?
- [x] Have the functional tests been executed?
- [x] Have all defects been reviewed and dispositioned, resolved, or deferred?
- [ ] Has the user story been accepted by the Product Owner?
- [x] Is the user story ready to be deployed to the test/staging environments?

**Comment by sdhawan16** [comment]
Created PR - moved code from MDM to Account API

**Comment by JanellC** [comment]
TC168 passed @tlageman-epa 


https://easey-dev.app.cloud.gov/api/account-mgmt/accounts


![image.png](https://images.zenhubusercontent.com/5f5928c1045cffdce7ce89c3/965edf36-d29d-4555-ab70-227ab07629e2)
